platform/ios/src/MGLMapView.mm

```diff
@@ -1169,6 +1169,10 @@ - (void)touchesBegan:(__unused NS_SET_OF(UITouch *) *)touches withEvent:(__unuse
 {
     _changeDelimiterSuppressionDepth = 0;
     _mbglMap->setGestureInProgress(false);
+    if (self.userTrackingState == MGLUserTrackingStateBegan)
+    {
+        [self setUserTrackingMode:MGLUserTrackingModeNone animated:NO];
```

Comment:
I agree, it's not incompatible. If you are pinching or rotating, the user tracking state should be possible or changed. In this case the tracking mode will not be changed. This code still works in all cases.
When the user changes the tracking mode, if an animation is needed, the tracking state goes to began as long as the flyTo completion is called. In any cases, if the user touch the screen while animating the map (to go to user location), it means he wants to cancel the move (and you to so by calling cancelTransistions()). Stopping the map where it is without returning to no tracking mode is a bug.
If the user is in tracking mode, pinching or rotating the map will not be affected by this code as the tracking state is not began.